lusSTR/tests/test_marker.py

```diff
@@ -265,8 +265,475 @@ def test_strobj_D21S11_newformat():
+    marker = STRMarkerObject('DXS10135', sequence, uas=False, kit='forenseq')
+    assert marker.annotation == bracketed
+    assert str(marker.canonical) == conc
+    assert marker.designation == (lus, sec, tert)
+
+
+@pytest.mark.parametrize('sequence, bracketed, conc, lus, sec, tert, flank_5p, flank_3p', [
+    (
+        'TGTGTGTGCATGCATACACACACAGAGAGAGAGAGAGAGAAAAAGAAGAAAGAAAGAAAGAAAGAAAGAAAGAAAGAAAGAAAGAAAG'
+        'AAAGAAAGAAAGAAAGAAAGGAAGAAAGAAAGGAAGAAAATAGAACAAATCAGCTTATATTCAGTATTTTTTTAGTATTTTCTGTGTC'
+        'AGCTC', '[AAGA]15 AAGG [AAGA]2', '18', '15', '0', None, '[TGTG]2 CATG CATA CACA CA C '
+        '[AGAG]4 AAAA AG', 'AAGG A AAGG AAGA AAAT AGAA CAAA TCAG CTTA TATT CAGT ATTT TTTT AGTA '
+        'TTTT CTGT GTCA G TC'
+    ),
+    (
+        'TGTGTGTGCATGCATACACACACAGAGAGAGAGAGAGAAAAAGAAGAAAGAAAGAAAGAAAGAAAGAAAGAAAGAAAGAAAGAAAGAA'
+        'AGAAAGAAAGAAAGAAAGGAAGAAAGAAAGGAAGAAAATAGAACAAATCAGCTTATATTCAGTATTTTTTTAGTATTTTCTGTGTCAG'
+        'CTC', 'GA [AAGA]14 AAGG [AAGA]2', '17.2', '14', '0', None, '[TGTG]2 CATG CATA CACA CA C '
+        '[AGAG]3 AGAA AAAG AA', 'AAGG A AAGG AAGA AAAT AGAA CAAA TCAG CTTA TATT CAGT ATTT TTTT '
+        'AGTA TTTT CTGT GTCA G TC'
+    )
+])
+def test_strobj_DXS10074(sequence, bracketed, conc, lus, sec, tert, flank_5p, flank_3p):
+    marker = STRMarkerObject('DXS10074', sequence, uas=False, kit='forenseq')
+    assert marker.annotation == bracketed
+    assert str(marker.canonical) == conc
+    assert marker.designation == (lus, sec, tert)
+    assert marker.flank_5p == flank_5p
+    assert marker.flank_3p == flank_3p
+
+
+@pytest.mark.parametrize('sequence, bracketed, conc, lus, sec, tert, kit', [
+    (
+        'CTATCTATCTATCTATTCATCCATCTAATCTATCCATTCTATCTATCTATCTATCTATCTATCTATCTATCTATCTATCTACCTACCT'
+        'ACCTATCTATCTATAGATCTATCTATCTATCT', 'C [TATC]3 TATT [CATC]2 TAAT CTAT CCAT [TCTA]11 '
+        '[CCTA]3 [TCTA]2 TAGA [TCTA]3 TCT', '11', '11', None, None, 'forenseq'
+    ),
+    (
+        'CATTCTATCTATCTATCTATCTATCTATCTATCTATCTATCTATCTACCTACCTACCTATCTATCTATAGATCTATCTATCTATCTTA'
+        'AATTTGGAAATTCTCCTCAGCATAACATTTTAATGATGATTCCTAGGATACAAGTGATGTGCTGAAAGTATCAATGTGTATCAGAAAA'
+        'CCAACATCTCTGCTTAGGTCTC', 'CAT [TCTA]11 CCTA CCTA CCTA [TCTA]2 TAGA [TCTA]3 TCT', '11',
+        '11', None, None, 'powerseq'
+    ),
+    (
+        'CATTCTATCTATCTATCTATCCATCTATCTATCTATCTATCTATCTACCTACCTACCTATCTATCTATAGATCTATCTATCTATCTTA'
+        'AATTTGGAAATTCTCCTCAGCATAACATTTTAATGATGATTCCTAGGATACAAGTGATGTGCTGAAAGTATCAATGTGTATCAGAAAA'
+        'CCAACATCTCTGCTTAGGTCTC', 'CAT [TCTA]4 TCCA [TCTA]6 CCTA CCTA CCTA [TCTA]2 TAGA [TCTA]3 '
+        'TCT', '11', '6', None, None, 'powerseq'
+    )
+])
+def test_strobj_Y_GATA_H4(sequence, bracketed, conc, lus, sec, tert, kit):
+    marker = STRMarkerObject('Y-GATA-H4', sequence, uas=False, kit=kit)
+    assert marker.annotation == bracketed
+    assert str(marker.canonical) == conc
+    assert marker.designation == (lus, sec, tert)
```

Comment:
In contrast, these tests do *a lot* to illuminate the reader's understanding. Kudos! I'm sure putting together these tests was tedious, and they'll be super valuable if any code refactoring is needed in the future.